openedx/core/djangoapps/content/learning_sequences/api/processors/enrollment.py

```diff
@@ -0,0 +1,38 @@
+"""
+Simple OutlineProcessor that removes items based on Enrollment and course visibility setting.
+"""
+from student.models import CourseEnrollment
+from .base import OutlineProcessor
+
+
+class EnrollmentOutlineProcessor(OutlineProcessor):
+    """
+    Simple OutlineProcessor that removes items based on Enrollment and course visibility setting.
+    """
+    def usage_keys_to_remove(self, full_course_outline):
```

Comment:
`EnrollmentOutlineProcessor` also needs to implement `inaccessible_sequences`, since a course with `public_outline` visibility should not have any clickable links.

Reply:
@ormsbee I don't get this point. We don't have any link in course outline yielded by outline API. So, what are you referring to?

Reply:
Sorry, that was unclear. You're right that we don't send URLs through the REST API. But we do send across an `accessible` attribute for every Sequence, which the front-end code knows to interpret to determine whether it should create clickable links (along with whatever logic it uses to map `sequence_ids` to actual URLs).

In any case, the method you implemented below looks fine.

Reply:
@ormsbee It was missing enrollment check i.e It should be
```python
        if full_course_outline.course_visibility == CourseVisibilities.PUBLIC_OUTLINE and not CourseEnrollment.is_enrolled(self.user, self.course_key):
```
instead of
```python
        if full_course_outline.course_visibility == "public_outline":
```

But, no worry I fixed it.